src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -27,6 +61,58 @@ public TypeSyntaxSimplifierWalker(CSharpSimplifyTypeNamesDiagnosticAnalyzer anal
             _semanticModel = semanticModel;
             _optionSet = optionSet;
             _cancellationToken = cancellationToken;
+
+            var root = semanticModel.SyntaxTree.GetRoot(cancellationToken);
+            _aliasedNames = GetAliasedNames(root as CompilationUnitSyntax);
+        }
+
+        private static ImmutableHashSet<string> GetAliasedNames(CompilationUnitSyntax? compilationUnit)
+        {
+            var aliasedNames = s_emptyAliasedNames;
```

Comment:
could just be a hashset as well.

Reply:
:arrow_right: see https://github.com/dotnet/roslyn/pull/41079#discussion_r369199277